clusterloader2/testing/load/config.yaml

```diff
@@ -477,283 +320,98 @@ steps:
-{{if not $DISABLE_DAEMONSETS}}
-  - namespaceRange:
-      min: 1
-      max: 1
-    replicasPerNamespace: 1
-    tuningSet: RandomizedScalingTimeLimited
-    objectBundle:
-      - basename: daemonset
-        objectTemplatePath: daemonset.yaml
-        templateFillMap:
-          Image: k8s.gcr.io/pause:3.1
-{{end}}
-  - namespaceRange:
-      min: 1
-      max: {{$namespaces}}
-    replicasPerNamespace: 1
-    tuningSet: RandomizedScalingTimeLimited
-    objectBundle:
-      - basename: small-job
-        objectTemplatePath: job.yaml
-        templateFillMap:
-          ReplicasMin: {{MultiplyInt $SMALL_GROUP_SIZE 0.5}}
-          ReplicasMax: {{MultiplyInt $SMALL_GROUP_SIZE 1.5}}
-  - namespaceRange:
-      min: 1
-      max: {{$namespaces}}
-    replicasPerNamespace: 1
-    tuningSet: RandomizedScalingTimeLimited
-    objectBundle:
-      - basename: medium-job
-        objectTemplatePath: job.yaml
-        templateFillMap:
-          ReplicasMin: {{MultiplyInt $MEDIUM_GROUP_SIZE 0.5}}
-          ReplicasMax: {{MultiplyInt $MEDIUM_GROUP_SIZE 1.5}}
-  - namespaceRange:
-      min: 1
-      max: {{$namespaces}}
-    replicasPerNamespace: {{$bigJobsPerNamespace}}
-    tuningSet: RandomizedScalingTimeLimited
-    objectBundle:
-      - basename: big-job
-        objectTemplatePath: job.yaml
-        templateFillMap:
-          ReplicasMin: {{MultiplyInt $BIG_GROUP_SIZE 0.5}}
-          ReplicasMax: {{MultiplyInt $BIG_GROUP_SIZE 1.5}}
+- module:
+    path: /modules/reconcile-objects.yaml
+    params:
+      actionName: "scaling and updating"
+      namespaces: {{$namespaces}}
+      randomScaleFactor: 0.5
+      daemonSetImage: k8s.gcr.io/pause:3.1
+      daemonSetReplicas: 1
+      bigDeploymentSize: {{$BIG_GROUP_SIZE}}
+      bigDeploymentsPerNamespace: {{$bigDeploymentsPerNamespace}}
+      mediumDeploymentSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumDeploymentsPerNamespace: {{$mediumDeploymentsPerNamespace}}
+      smallDeploymentSize: {{$SMALL_GROUP_SIZE}}
+      smallDeploymentsPerNamespace: {{$smallDeploymentsPerNamespace}}
+      smallStatefulSetSize: {{$SMALL_GROUP_SIZE}}
+      smallStatefulSetsPerNamespace: {{$SMALL_STATEFUL_SETS_PER_NAMESPACE}}
+      mediumStatefulSetSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumStatefulSetsPerNamespace: {{$MEDIUM_STATEFUL_SETS_PER_NAMESPACE}}
+      bigJobSize: {{$BIG_GROUP_SIZE}}
+      bigJobsPerNamespace: 1
+      mediumJobSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumJobsPerNamespace: 1
+      smallJobSize: {{$SMALL_GROUP_SIZE}}
+      smallJobsPerNamespace: 1
 
-- name: Waiting for objects to become scaled
-  measurements:
-  - Method: WaitForControlledPodsRunning
-    Instances:
-    - Identifier: WaitForRunningDeployments
-    - Identifier: WaitForRunningStatefulSets
-    - Identifier: WaitForRunningDaemonSets
-    - Identifier: WaitForRunningJobs
-    Params:
-      action: gather
+- module:
+    path: /modules/reconcile-objects.yaml
+    params:
+      actionName: "deleting"
```

Comment:
In other places we use `action: gather` or `action: start`. I think we could rename this to delete/create etc.

Reply:
Done.